### PR TITLE
[Sun Yucheng] iP

### PR DESCRIPTION
**Pull request by Cheng20010201**

### CODE REVIEW

File: src/main/java/lihua/commands/FindCommand.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package lihua.commands;
+
+/**
+ * Command class representing a command to find tasks by keyword.
+ */
+public class FindCommand extends Command {
+    /** Command word for find command */
```
Comment (ZhangAnli) [inline src/main/java/lihua/commands/FindCommand.java]: Do you think it would help if you left spaces between consecutive methods? Have noticed this in a few places and because of the javadoc comments, it may make your code seem a bit "cramped".

File: src/main/java/lihua/tasks/Deadline.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import org.json.simple.JSONObject;
+
+/**
+ * Represents a lihua.tasks.Deadline item, which is a child of lihua.tasks.Task
+ */
+public class Deadline extends Task {
+    private final LocalDate by;
```
Comment (ZhangAnli) [inline src/main/java/lihua/tasks/Deadline.java]: Perhaps a more intuitive name for the localdate variable will be better? I feel that just stating "by" could be hard to read for some.

File: src/test/java/lihua/tasks/DeadlineTest.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package lihua.tasks;
+
+public class DeadlineTest {
```
Comment (ZhangAnli) [inline src/test/java/lihua/tasks/DeadlineTest.java]: Would like to clarify the purpose of this file since it is essentially an empty file?

**Review by ZhangAnli — COMMENTED** [review]
Overall, the code was written well and did not violate much coding violations. Good job!

File: src/test/java/lihua/tasks/DeadlineTest.java (hunk 3 above)
Comment (Cheng20010201) [reply src/test/java/lihua/tasks/DeadlineTest.java]: Oops I added the test class but forgot to add content to it. Thanks for pointing it out!

File: src/main/java/lihua/tasks/Deadline.java (hunk 2 above)
Comment (Cheng20010201) [reply src/main/java/lihua/tasks/Deadline.java]: like `deadlineDate`? 🤔

File: src/main/java/lihua/commands/FindCommand.java (hunk 1 above)
Comment (Cheng20010201) [reply src/main/java/lihua/commands/FindCommand.java]: Sorry correct me if I misunderstood your point🤔. Are you suggesting like I leave spaces of more than one line between consecutive methods to make the code look more 'sparse'?

Comment (Eriksen2411) [reply src/main/java/lihua/commands/FindCommand.java]: For me, I think the spacing between consecutive methods is good enough. I just think that at the beginning of the class ( between attributes and constructor) you should leave a one line spacing for easier looking

File: src/main/java/lihua/commands/HelpCommand.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package lihua.commands;
+
+import lihua.commons.Messages;
+
+/**
+ * Command class representing a command to get help.
+ */
+public class HelpCommand extends Command {
+    /** Command word for help command */
+    public static final String COMMAND_WORD = "help";
```
Comment (Eriksen2411) [inline src/main/java/lihua/commands/HelpCommand.java]: Do we need COMMAND_WORD and MESSAGE_USAGE here at the same time both serving as attributes. I think you should only keep COMMAND_WORD as attribute and the other could be converted to method return.

File: src/main/java/lihua/tasks/Deadline.java (hunk 2 above)
Comment (Eriksen2411) [reply src/main/java/lihua/tasks/Deadline.java]: Hmmm i also agree with ZhangAnli this point. and deadlineDate seems good enough.

File: src/test/java/lihua/tasks/DeadlineTest.java (hunk 3 above)
Comment (Eriksen2411) [reply src/test/java/lihua/tasks/DeadlineTest.java]: Also agree :)) Just for me to add 1 comment. TY

File: src/main/java/lihua/tasks/Deadline.java (hunk 2 above)
Comment (branzuelajohn) [reply src/main/java/lihua/tasks/Deadline.java]: I agree with Anli and Eriksen regarding this

File: src/main/java/lihua/tasks/Task.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+     * Initialize a new undone task
+     * @param name Name of the task
+     */
+    public Task(String name) {
+        this.name = name;
+        isDone = false;
+    }
+
+    /**
+     * Set the status of the task; e.g. done or not done
```
Comment (branzuelajohn) [inline src/main/java/lihua/tasks/Task.java]: "Sets" instead of set?

File: src/test/java/lihua/tasks/EventTest.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package lihua.tasks;
+
+public class EventTest {
```
Comment (branzuelajohn) [inline src/test/java/lihua/tasks/EventTest.java]: Are you supposed to add content to this test class file, because this class does not serve a purpose now as it is empty

File: src/main/java/lihua/main/Main.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+import javafx.application.Application;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Scene;
+import javafx.scene.layout.AnchorPane;
+import javafx.stage.Stage;
+import lihua.ui.MainWindow;
+
+/**
+ * A GUI for Duke using FXML.
```
Comment (branzuelajohn) [inline src/main/java/lihua/main/Main.java]: should this be "A GUI for Lihua..." to be consistent with the name you decided to name your bot
